velox/expression/VectorUdfTypeSystem.h

```diff
@@ -359,14 +366,18 @@ struct VectorReader<Map<K, V>> {
   }
 
   bool containsNull(vector_size_t index) const {
+    VELOX_DCHECK(
+        keysMayHaveNulls_.has_value() && valuesMayHaveNulls_.has_value(),
+        "setChildrenMayHaveNulls() should be called before containsNull()");
+
     auto decodedIndex = decoded_.index(index);
 
     return decoded_.isNullAt(index) ||
-        (decodedKeys_.mayHaveNullsRecursive() &&
+        (keysMayHaveNulls_.value_or(true) &&
```

Comment:
consider the perf of this vs *setChildrenMayHaveNulls.

Reply:
I ran the benchmarks and I didn't see a significant perf change going from just a plain old bool to this.

given the VELOX_DCHECK only runs in debug mode, I'd prefer to keep it like this for safety given there's no big perf hit.